# #App Data Analysis
This is a small analysis project on various apps on android and ios inorder to find out trends among app users based on 2017 and 2018 data.Here i try to do an analysis on free apps and its popularity 

In [1]:
from csv import reader
#opening applestore data
df_apple_store=open('AppleStore.csv',encoding="utf8")
df_apple_read=reader(df_apple_store)
df_list_apple=list(df_apple_read)
df_header_apple=df_list_apple[0]
df_data_apple=df_list_apple[1:]
#opening google playstore data
df_googleplaystore=open('googleplaystore.csv',encoding="utf8")
df_google_read=reader(df_googleplaystore)
df_list_google=list(df_google_read)
df_header_google=df_list_google[0]
df_data_google=df_list_google[1:]


exporting data

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') 

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))


In [3]:
explore_data(df_data_google,0,6,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+

# DATA CLENSING

In [4]:
#it is findout that there is a missing data at index number 10472
#for the sake of accurate analysis that data is removed
del df_data_google[10472]


# Finding duplicate rows
Duplicated rows will reduce the  accuracy of our analysis.So we have to keep only one row for each app details.


In [5]:
unique_data=[]
duplicate_data=[]
for row in df_list_apple:
    name=row[0]
    if name in unique_data:
        duplicate_data.append(name)
    else:
        unique_data.append(name)
print(unique_data)
print(duplicate_data)#we get an empty list as result which means 
#there is no duplicate data


['id', '284882215', '389801252', '529479190', '420009108', '284035177', '429047995', '282935706', '553834731', '324684580', '343200656', '512939461', '362949845', '359917414', '469369175', '924373886', '575658129', '506627515', '500116670', '479516143', '293778748', '341232718', '440045374', '295646461', '487119327', '284815942', '596402997', '466965151', '293622097', '350642635', '582654048', '352683833', '543186831', '422667065', '457446957', '510461758', '284993459', '860822992', '307727765', '651510680', '708600202', '930574573', '526641427', '297368629', '304878510', '698255242', '488627858', '366247306', '333903271', '454638411', '323229106', '310738695', '305343404', '403858572', '387428400', '447188370', '363590051', '331177714', '640111933', '572395608', '421167112', '290638154', '307906541', '317469184', '310633997', '1153883316', '544007664', '561941526', '497595276', '597986893', '1053012308', '625334537', '282614216', '357218860', '436491861', '1094591345', '887947640', '8

In [6]:
unique_data=[]
duplicate_data=[]
for row in df_list_google:
    name=row[0]
    if name in unique_data:
        duplicate_data.append(name)
    else:
        unique_data.append(name)

print(duplicate_data)#we get 1181 duplicate 
                          #data on googleplaystore dataset
print(len(duplicate_data))


['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software', 'MailChimp - Email, Marketing Automation', 'Crew - Free Messaging and Scheduling', 'Asana: organize team projects', 'Google Analytics', 'AdWords Express', 'Accounting App - Zoho Books', 'Invoice & Time Tracking - Zoho', 'join.me - Simple Meetings', 'Invoice 2go — Professional Invoices and Estimates', 'SignEasy | Sign and Fill PDF and other Documents', 'Quick PDF Scanner + OCR FREE', 'Genius Scan - PDF Scanner', 'Tiny Scanner - PDF Scanner App', 'Fast Scanner : Free PDF Scan', 'Mobile Doc Scanner (MDScan) Lite', 'TurboScan: scan documents and receipts in PDF', 'Tiny Scanner Pro: PDF Doc Scan', 'Docs To Go™ Free Office Suite', 'OfficeSuite : Free Office + PDF Editor',

In [7]:
df_header_google

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

# Criterion for removing duplicate data
we get 1181 number of duplicate entries in googleplaystore data set.instead of removing randomly duplicate entries im going to keep most recently updated data and remove rest of the data 

Here by analysing the Reviews column we can understand the duplicated apps have different reviews value.It is because the data is collected in different time and the highest value of review shows the latast data.So for the recent data here i pick most reviewed data of duplicated apps and ignore rest of the data

In [8]:
reviews_max={}#creating a dictonary for storing highest review values for app keyword
for row in df_data_google:
    app_name=row[0]
    n_reviews=float(row[3])
    if app_name in reviews_max and reviews_max[app_name]<n_reviews:
        reviews_max[app_name]=n_reviews
    if app_name not in reviews_max:
        reviews_max[app_name]=n_reviews
        


In [9]:
print(len(reviews_max))

9659


In [10]:
print((reviews_max))

{'Photo Editor & Candy Camera & Grid & ScrapBook': 159.0, 'Coloring book moana': 974.0, 'U Launcher Lite – FREE Live Cool Themes, Hide Apps': 87510.0, 'Sketch - Draw & Paint': 215644.0, 'Pixel Draw - Number Art Coloring Book': 967.0, 'Paper flowers instructions': 167.0, 'Smoke Effect Photo Maker - Smoke Editor': 178.0, 'Infinite Painter': 36815.0, 'Garden Coloring Book': 13791.0, 'Kids Paint Free - Drawing Fun': 121.0, 'Text on Photo - Fonteee': 13880.0, 'Name Art Photo Editor - Focus n Filters': 8788.0, 'Tattoo Name On My Photo Editor': 44829.0, 'Mandala Coloring Book': 4326.0, '3D Color Pixel by Number - Sandbox Art Coloring': 1518.0, 'Learn To Draw Kawaii Characters': 55.0, 'Photo Designer - Write your name with shapes': 3632.0, '350 Diy Room Decor Ideas': 27.0, 'FlipaClip - Cartoon animation': 194216.0, 'ibis Paint X': 224399.0, 'Logo Maker - Small Business': 450.0, "Boys Photo Editor - Six Pack & Men's Suit": 654.0, 'Superheroes Wallpapers | 4K Backgrounds': 7699.0, 'Mcqueen Color

# removing duplicated rows

In [11]:
android_clean = []
already_added = []

for app in df_data_google:
    app_name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[app_name] == n_reviews) and (app_name not in already_added):
        android_clean.append(app)
        already_added.append(app_name)

In [12]:
print(android_clean)

[['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up'], ['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up'], ['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up'], ['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up'], ['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up'], ['Smoke Effect Photo Maker - Smoke Editor', 'ART_AND_DESIGN', '3.8', '178', '19M', '50,000+', '

In [13]:
print(already_added)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'Sketch - Draw & Paint', 'Pixel Draw - Number Art Coloring Book', 'Paper flowers instructions', 'Smoke Effect Photo Maker - Smoke Editor', 'Infinite Painter', 'Garden Coloring Book', 'Kids Paint Free - Drawing Fun', 'Text on Photo - Fonteee', 'Name Art Photo Editor - Focus n Filters', 'Tattoo Name On My Photo Editor', 'Mandala Coloring Book', '3D Color Pixel by Number - Sandbox Art Coloring', 'Learn To Draw Kawaii Characters', 'Photo Designer - Write your name with shapes', '350 Diy Room Decor Ideas', 'FlipaClip - Cartoon animation', 'ibis Paint X', 'Logo Maker - Small Business', "Boys Photo Editor - Six Pack & Men's Suit", 'Superheroes Wallpapers | 4K Backgrounds', 'HD Mickey Minnie Wallpapers', 'Harley Quinn wallpapers HD', 'Colorfit - Drawing & Coloring', 'Animated Photo Editor', 'Pencil Sketch Drawing', 'Easy Realistic Drawing Tutorial', 'Pink Silver Bow Keyboard Theme', 'Art Dr

# Removing non english app names from list

In [14]:
def is_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii > 3: #ASCII value of english charactors are below 127.if more than 3 non english charactors are detected 
        return False #remove that app from analysis
    else:
        return True

print(is_english('Docs To Go™ Free Office Suite'))#only have one non-english charactor.so it takes into account
print(is_english('Instachat 😜'))#only have one non-english charactor.so it takes into account

True
True


In [15]:
android_english = []
ios_english = []

for app in android_clean:#iterate through anaroid apps for finding english app names only 
    name = app[0]
    if is_english(name):
        android_english.append(app)
        
for app in df_data_apple:#iterate through ios apps for finding english app names only 
    name = app[1]
    if is_english(name):
        ios_english.append(app)
        
explore_data(android_english, 0, 3, True)
print('\n')
explore_data(ios_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

# Analysing  on free apps 

In [16]:
android_final = []
ios_final = []

for app in android_english:#iterate through android apps for finding free apps only
    price = app[7]
    if price == '0':
        android_final.append(app)
        
for app in ios_english:#iterate through ios apps for finding free apps only
    price = app[4]
    if price == '0.0':
        ios_final.append(app)
        
print(len(android_final))
print(len(ios_final))

8864
3222


# Most common app by genre

For this im creating a frequency table by genre.Below there is a function which takes dataset and index number for creating the frequency table.Here i use python dictonary data type as frequency table.

In [25]:
df_header_google

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

In [29]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1 #counting total number of data
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    
    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100 #finding percentage
        table_percentages[key] = percentage  #adding percentage value to key in new dictonary
    
    return table_percentages


def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)#creating tuple data type from dictonary data type
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)#sorting in ascending order
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

# for Andriod

In [30]:
display_table(android_final,-4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

From this result we can conclude that in android apps most apps are under genre of tools(8%).and second position is for entertainment.The above list is sorted in acending order so we can see very least apps are for Adventure and educational apps.

# For IOS

In [19]:


display_table(ios_final, -5)



Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


in case of ios apps the genres are few compare with android.However more than half percentage of apps are for entertainment(58%).other genres are just below 10%.We can see a huge gap between percentages

Anyway from both app genre analysis we can see entertainment is always shows a good amount of quantity among both ios and android apps.

# Most popular app by genre in ios

Here we going to check popularity of apps by counting number of installs by users.It will show us which genre is more popular among users 

In [20]:
df_header_apple

['id',
 'track_name',
 'size_bytes',
 'currency',
 'price',
 'rating_count_tot',
 'rating_count_ver',
 'user_rating',
 'user_rating_ver',
 'ver',
 'cont_rating',
 'prime_genre',
 'sup_devices.num',
 'ipadSc_urls.num',
 'lang.num',
 'vpp_lic']

from header of ios dataset we can use the total number of user rating(rating_count_tot) as total number of users and find out most popular app in ios.note that this is an assumption and not going to give auctual values of total number of installs of that particular app.

In [32]:
genres_ios = freq_table(ios_final, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:            
            n_ratings = float(app[5])
            total += n_ratings #adding number of user reviews together 
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)


Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


from this result we can see more number of user reviews on Navigation genre.From this we can assume that Navigation genre is more popular among users in ios.Medical genre is very less popular among ios genres

Lets analyse close on navigation

In [36]:
for app in ios_final:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


On navigation genre we can see most rated app is Waze - GPS Navigation, Maps & Real-time Traffic. and second position is for google map

# Most popular app by genre in android


In [22]:
df_header_google

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

Here installs is directly given.so we can find frequency table directly from our used defined function which we created in In[44] to find percentage of each install numbers.However the number is not accurate because it just dont give us auctual number of installs just give eg (10000+) rough figures.Anyway we going to take it as 10000(there is no way).To get a error free code we should replace '+' sign and ',' by just '' and convert the value to float for add up . After that we can easily addup the number of installs based on catagory on android apps. 

In [23]:
display_table(android_final, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


Number of installs based on catagory

In [24]:
categories_android = freq_table(android_final, 1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:            
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

By analysing communication apps are mostly installed on android app (38456119).

Lets analyse closely on communication apps which are top installed

In [38]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts are most popular apps in communication genre